textdb/textdb-textql/src/test/java/edu/uci/ics/textdb/textql/languageparser/TextQLParserTest.java

```diff
@@ -0,0 +1,616 @@
+    @Test
+    public void testNumberLiteral() throws ParseException {
+        Assert.assertEquals((new TextQLParser(string2InputStream(" 0 "))).numberLiteralToString(), "0");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" 12 "))).numberLiteralToString(), "12");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" 34566 "))).numberLiteralToString(), "34566");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" 78.90 "))).numberLiteralToString(), "78.90");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" 123. "))).numberLiteralToString(), "123.");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" .456 "))).numberLiteralToString(), ".456");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" -0 "))).numberLiteralToString(), "-0");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" -12 "))).numberLiteralToString(), "-12");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" -34566 "))).numberLiteralToString(), "-34566");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" -78.90 "))).numberLiteralToString(), "-78.90");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" -123. "))).numberLiteralToString(), "-123.");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" -.456 "))).numberLiteralToString(), "-.456");
+        Assert.assertEquals((new TextQLParser(string2InputStream(" -.789.001 "))).numberLiteralToString(), "-.789");
+        assertException(()->(new TextQLParser(string2InputStream(" -e "))).numberLiteralToString(), TokenMgrError.class);
+        assertException(()->(new TextQLParser(string2InputStream(" -e 21"))).numberLiteralToString(), TokenMgrError.class);
+        assertException(()->(new TextQLParser(string2InputStream(" +4 "))).numberLiteralToString(), TokenMgrError.class);
+        assertException(()->(new TextQLParser(string2InputStream(" a "))).numberLiteralToString(), ParseException.class);
+        assertException(()->(new TextQLParser(string2InputStream(" a 22 "))).numberLiteralToString(), ParseException.class);
+        assertException(()->(new TextQLParser(string2InputStream(" a45 "))).numberLiteralToString(), ParseException.class);
+        assertException(()->(new TextQLParser(string2InputStream(" A45 "))).numberLiteralToString(), ParseException.class);
+        assertException(()->(new TextQLParser(string2InputStream(" FROM45 "))).numberLiteralToString(), ParseException.class);
+        assertException(()->(new TextQLParser(string2InputStream(" \"4\" "))).numberLiteralToString(), ParseException.class);
+        assertException(()->(new TextQLParser(string2InputStream(" /4/ "))).numberLiteralToString(), ParseException.class);
+        assertException(()->(new TextQLParser(string2InputStream(" /4 "))).numberLiteralToString(), TokenMgrError.class);
+        
+    }
+    
+    @Test
+    public void testNumberDouble() throws ParseException {
```

Comment:
Add a high-level comment to explain the purpose of this case and its main logic.